grails-app/services/com/unifina/service/EthereumIntegrationKeyService.groovy

```diff
@@ -32,9 +33,26 @@ class EthereumIntegrationKeyService {
 
 	@PostConstruct
 	void init() {
-		String password = grailsApplication.config["streamr"]["encryption"]["password"]
+		initStringEncryptor()
```

Comment:
I'm actually wondering why the changes in this class were necessary? What was the problem before the changes?

Reply:
When I revert it to the original, it fails to load the password correctly and decrypt key in IntegrationKeySpec:24

`| Failure:  toMap().json for Ethereum integration key(com.unifina.domain.security.IntegrationKeySpec)
|  java.lang.NullPointerException: Cannot invoke method decrypt() on null object
	at com.unifina.service.EthereumIntegrationKeyService.decryptPrivateKey(EthereumIntegrationKeyService.groovy:128)
	at com.unifina.domain.security.IntegrationKey.jsonMap(IntegrationKey.groovy:55)
	at com.unifina.domain.security.IntegrationKey.toMap(IntegrationKey.groovy:45)
	at com.unifina.domain.security.IntegrationKeySpec.toMap().json for Ethereum integration key(IntegrationKeySpec.groovy:24)
`
I remember wrestling with this months ago. Beans are instantiated differently in test vs production. I think @PostConstruct is ignored when testing or something like that. I remember init() wasn't automatically called in test. I don't understand grails well enough to say exactly why.

Reply:
Ok so to keep things reliable, let's delete the `init()` function and get rid of `@PostConstruct` completely. It's ok to initialize `encryptor` lazily, like you're doing in `getStringEncryptor()`. All access to `encryptor` should be via `getStringEncryptor()` anyway.